src/chain_sync/network_context.rs

```diff
@@ -290,13 +294,18 @@ where
                 // a request succeeds.
                 let peers = self.peer_manager.top_peers_shuffled();
                 let mut batch = RaceBatch::new(MAX_CONCURRENT_CHAIN_EXCHANGE_REQUESTS);
+                let success_time_cost_millis_total = Arc::new(AtomicU64::new(0));
+                let success_count = Arc::new(AtomicU64::new(0));
```

Comment:
Could we abstract it to a higher level class to avoid having this lower-level logic in such places?

Reply:
Does this crate look like a good fit? https://crates.io/crates/incr_stats

Reply:
I'd avoid relying on niche crates, since it should be simple to implement a basic abstraction ourselves. Niche crates bloat dependency tree and are more susceptible to malicious code (since there's hardly anyone reviewing the changes).

Reply:
Fixed.